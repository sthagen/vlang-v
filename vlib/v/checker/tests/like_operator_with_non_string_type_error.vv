struct User {
	id   int    [primary]
	name string
}

fn main() {
	db := ''

	println(sql db {
		select from User where name like 10
	}!)

	println(sql db {
		select from User where 10 like true
	}!)
}
