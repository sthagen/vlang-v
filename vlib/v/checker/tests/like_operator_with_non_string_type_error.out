vlib/v/checker/tests/like_operator_with_non_string_type_error.vv:10:36: error: the right operand of the `like` operator must be a string type
    8 | 
    9 |     println(sql db {
   10 |         select from User where name like 10
      |                                          ~~
   11 |     }!)
   12 |
vlib/v/checker/tests/like_operator_with_non_string_type_error.vv:14:26: error: the left operand of the `like` operator must be an identifier with a string type
   12 | 
   13 |     println(sql db {
   14 |         select from User where 10 like true
      |                                ~~
   15 |     }!)
   16 | }
